tests/tests_unit/test_data_classes/test_data_models/test_filters.py

```diff
@@ -250,6 +250,31 @@ def dump_filter_test_data() -> Iterator[ParameterSet]:
     )
     yield pytest.param(prop_list1, expected, id="Prefix filter with list property of objects")
     yield pytest.param(prop_list2, expected, id="Prefix filter with list property of dicts")
+    overloaded_filter = f.Equals(property="name", value="bob") & f.HasData(
+        containers=[("space", "container")]
+    ) | ~f.Range(property="size", gt=0)
+    expected = {
+        "or": [
+            {
+                "and": [
+                    {"equals": {"property": ["name"], "value": "bob"}},
+                    {"hasData": [{"type": "container", "space": "space", "externalId": "container"}]},
+                ]
+            },
+            {"not": {"range": {"property": ["size"], "gt": 0}}},
+        ]
+    }
+    yield pytest.param(overloaded_filter, expected, id="Compound filter with overloaded operators")
+    nested_overloaded_filter = (f.Equals("a", "b") & f.Equals("c", "d")) & (f.Equals("e", "f") & f.Equals("g", "h"))
+    expected = {
+        "and": [
+            {"equals": {"property": ["a"], "value": "b"}},
+            {"equals": {"property": ["c"], "value": "d"}},
+            {"equals": {"property": ["e"], "value": "f"}},
+            {"equals": {"property": ["g"], "value": "h"}},
+        ]
+    }
+    yield pytest.param(nested_overloaded_filter, expected, id="Compound filter with nested overloaded and")
```

Comment:
Could also test that two CompundFilters combine correctly, e.g.:
```
f1, f2, f3, f4, f5 = map(Exists, "abcde")

ff1 = (And(f1) & And(f2)) & (And(f3) & (And(f4) & And(f5)))
ff2 = And(f1) & And(f2) & And(f3) & And(f4) & And(f5)

expected = {'and': [{'exists': {'property': ['a']}},
  {'exists': {'property': ['b']}},
  {'exists': {'property': ['c']}},
  {'exists': {'property': ['d']}},
  {'exists': {'property': ['e']}}]}

assert expected == ff1.dump()
assert ff1.dump() == ff2.dump()

Reply:
Is this to take care of cases where the operators are not commutative?

Reply:
Ah no, was thinking of the case when both left and right filters are of the given compound filter type in `_list_filters_without_nesting`